### PR TITLE
refactor: deprecate child bindings apply options

### PR DESCRIPTION
**Pull request by climba03003**
Deprecate apply options through bindings for child logger.

### CODE REVIEW

File: benchmarks/child-creation.bench.js [hunk 1]
```diff
@@ -60,6 +60,13 @@ const run = bench([
       child.info({ hello: 'world' })
     }
     setImmediate(cb)
+  },
+  function benchPinoCreationWithOption (cb) {
+    const child = plogDest.child({ a: 'property' }, { redact: [] })
+    for (var i = 0; i < max; i++) {
```
Comment (kibertoad) [inline benchmarks/child-creation.bench.js:66]: why not let?

File: browser.js [hunk 2]
```diff
@@ -113,13 +113,17 @@ function pino (opts) {
...
-    const bindingsSerializers = bindings.serializers
-    if (serialize && bindingsSerializers) {
-      var childSerializers = Object.assign({}, serializers, bindingsSerializers)
+    childOptions = childOptions || {}
+    if (serialize && bindings.serializers) {
+      childOptions.serializers = bindings.serializers
+    }
+    const childOptionsSerializers = childOptions.serializers
+    if (serialize && childOptionsSerializers) {
+      var childSerializers = Object.assign({}, serializers, childOptionsSerializers)
```
Comment (kibertoad) [inline browser.js:126]: why not let?

**Comment by kibertoad** [comment]
@climba03003 Was this intended for master and not next?

**Comment by climba03003** [comment]
> 
> 
> @climba03003 Was this intended for master and not next?

It should comes with #1059 and nothing is breaking. So, this PR can be landed on both `master` and `next` without problem.

File: browser.js (hunk 2 above)
Comment (climba03003) [reply browser.js:126]: Do you means `var childSerializers` here? I do not know why it is `var` in the beginning, so I will keep it.

File: benchmarks/child-creation.bench.js (hunk 1 above)
Comment (climba03003) [reply benchmarks/child-creation.bench.js:66]: the looping is copy from other test. As I remember, `for-var` loop out perform `for-let` loop. I think that is the reason.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by kibertoad** [comment]
@climba03003 next likely needs a slight TS types updates

**Review by kibertoad — APPROVED** [review]

File: browser.js (hunk 2 above)
Comment (kibertoad) [reply browser.js:126]: yes. yeah, it's a nitpick, just no good reason to use var anymore, so eventually we should convert.

File: benchmarks/child-creation.bench.js (hunk 1 above)
Comment (kibertoad) [reply benchmarks/child-creation.bench.js:66]: huh, interesting. TIL

**Comment by climba03003** [comment]
One of the fields is missing `bindings.level`. I am thinking of, how to test the deprecate warning.

**Comment by kibertoad** [comment]
@climba03003 How about putting a spy on console? might need an additional lib, though

**Comment by climba03003** [comment]
> 
> 
> doesn't this also need an assert to check the overall amount of asserts in test to ensure this is called?

`deprecations.test.js` is a tricky test files, it all rely on `pass` to prevent error. If we use the fixed test-case amount. It will becomes all failed. As the `onWarning` functions will receive all the other warning at the same time.

**Review by jsumners — APPROVED** [review]

**Pull request merged**